articles/how-to-use-local-storage-with-JavaScript/index.md

```diff
@@ -0,0 +1,248 @@
+
+### Main methods in localStorage
+The primary functions when using localStorage are `key()`, `setItem()`, `removeItem()`, `getItem()`, and `clear()`.
+
+#### key()
+This function is used to retrieve a value/string in a specific location. The index can be passed into the key() function as a parameter.
+
+```javascript
+var answer = localStorage.key(1);
+
+// this statement will retrieve the value of the second item in the localStorage.
+```
+
+The key() can also be used in a for-loop to retrieve all the items in the localStorage.
+
+#### setItem()
+This function is used to store items in the localStorage. An example of this function is shown below.
+
+```
+Window.localStorage.setItem("grade","One");
+
+//in this case, the `grade` is the key while `One` is the value.
+```
+
+As noted, we must `stringify` objects before we store them in the localStorage. An example is outlined below.
+
+```javascript
+Const Car = {
+
+brand:"Suzuki",
+
+color:"white",
+
+price:10000
+
+}
+
+Window.localStorage.setItem('car', JSON.stringify(Car));
+```
+
+Failure to stringify the object will result in an error.
+
+#### getItem()
+This function enables one to access or retrieve the data in the localStorage. The method takes in a `key` as a parameter. It then extracts the required value.
+
+For example, to retrieve the above Car object, we will use the following statement.
+
+```
+Window.localStorage.getItem('car');
+```
+
+The above statement will return something like this;
+
+```javascript
+"{brand:"Suzuki",color:"white",price:"10000"}"
+```
+
+To make maximum use of the above result, you should convert it to an object using `JSON.parse()`.
+
+```
+JSON.parse(window.localStorage.getItem('car';));
+```
+
+#### removeItem()
+This method is used to delete an item from the localStorage. The `removeItem()` method requires a key as a parameter. This is shown below.
+
+```
+window.localStorage.removeItem('brand');
+```
+
+#### clear()
+When called, this function clears all values in the localStorage. It is executed, as shown below. It does not require any parameters.
+
+```javascript
+Window.localStorage.clear()
+```
+
+Now that we have understood what localStorage entails let&#39;s create a project that uses this concept.
+
+### Preferences
+You must have a basic understanding of javascript. You also need a code editor and browser to test the project. In this example, we will mainly be using visual studio code and Chrome.
+
+### Goal
+To create a web application that stores, retrieves, deletes, and clears items from the `localStorage`.
+
+Let&#39;s jump in.
+
+### 1. Creating the project
+Create a new folder on your Desktop and open it in your code editor. Create two files: `index.html` and `main.js`. The `index.html` will showcase the web page to the user, while the `main.js` will store our javascript functions. These methods will be used to access different functionalities of the `localStorage`.
```

Comment:
These methods will be used to access different functionalities of the `localStorage`.

You must say functions, not methods,